src/Halcyon/HAL/HALResponse.cs

```diff
@@ -26,9 +25,9 @@ public class HALResponse {
         public HALResponse(object model, IHALModelConfig config = null)
             : this(config) {
             if (!(model is JObject) && (model is IEnumerable)) {
-                throw new ArgumentException("The HAL model should be Enumerable. You should use an embedded collection instead", "model");
+                throw new ArgumentException("The HAL model should be Enumerable. You should use an embedded collection instead", nameof(model));
             }
-
+            HALAttributeResolver.ResolveAttributes(this, model);
```

Comment:
Why do you need to make the private field internal?

I can see your point about the pagination scenario. I'll have to think on that.